src/syzygy/syzygy.cc

```diff
@@ -1047,6 +1047,11 @@ class SyzygyTablebaseImpl {
     }
     *mapping = statbuf.st_size;
     base_address = mmap(nullptr, statbuf.st_size, PROT_READ, MAP_SHARED, fd, 0);
+#if defined(MADV_RANDOM)
+    // For context: <https://github.com/official-stockfish/Stockfish/pull/1829>
+    // and <https://github.com/official-stockfish/Stockfish/pull/3094>.
+    madvise(base_address, statbuf.st_size, MADV_RANDOM);
+#endif
```

Comment:
Do you want to add the random access hint to the windows code path as well?

Reply:
You mean `FILE_FLAG_RANDOM_ACCESS `?

Reply:
yeah